C++ compilation failing on alpine

### Version

- Ice version [3.8.0]
- OS: Linux
- Distribution: Alpine Linux 3.22.1

### Describe the bug

pip install zeroc-ice is failing due to a build error in the c++ backend.
Compilation is failing due to execinfo.h not being found:
```
      c++ -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -DICE_DISABLE_PRAGMA_COMMENT -DICE_API=/**/ -DICE_DISCOVERY_API=/**/ -DICE_LOCATOR_DISCOVERY_API=/**/ -Idist/python/modules/IcePy -Idist/ice/cpp/include -Idist/ice/cpp/include/generated -Idist/ice/cpp/src -Idist/mcpp -I/usr/include/python3.12 -c dist/ice/cpp/src/Ice/LocalException.cpp -o build/temp.linux-x86_64-cpython-312/dist/ice/cpp/src/Ice/LocalException.o -std=c++17 -w
      dist/ice/cpp/src/Ice/LocalException.cpp:53:18: fatal error: execinfo.h: No such file or directory
         53 | #        include <execinfo.h>
            |                  ^~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/c++' failed with exit code 1
```

execinfo was removed from alpine:
https://gitlab.alpinelinux.org/alpine/aports/-/commit/50795a14dee639ce2dcc836e2b2baca9bad4a1b1

### Expected behavior

pip-install succeeds to build the c++ backend.

### Additional context

_No response_

--- a/cpp/src/Ice/LocalException.cpp
+++ b/cpp/src/Ice/LocalException.cpp
@@ -48,7 +48,7 @@
 #        endif
 #    endif
 
-#    if !defined(__FreeBSD__)
+#    if __has_include(<execinfo.h>)
 #        include <cstdint>
 #        include <execinfo.h>
 #        define ICE_BACKTRACE

